### DISCUSSION TITLE
How to quickly put the nc file created by myself into MET for forecast evaluation

### DISCUSSION THREAD

**Discussion by ZhiboGao** [dtcenter/METplus#1088] [File I/O] [answered]
Dear wonderful MET help team,

My name is Zhibo Gao, a PhD student from Beijing Normal University and also a MET user. I am very glad to write to you. First of all, thank you for developing such a great tool！ When I use this tool, I encounter some confusion, I would appreciate if you could help me.

I currently get a forecasted two-dimensional precipitation field named rain_forecast ([lat | 87] x [lat | 123]), correspondingly, I have an observed precipitation variable named rain_obs ([lat | 87] x [lon | 123]), both of these variables come from the nc file generated by myself, which may not meet the file standard of MET. I want to know if there is an easy way to make MET evalutate these two variables ( For example, using spatial verification methods).

By the way, if these two variables are rain_forecast ([lat | 87] x [time| 123]) and rain_obs ([lat | 87] x [time | 123]), is there a simple way to make MET evalutate these two variables ( For example, also using spatial verification methods).

Thank you for your help. 
Kind regards,
Zhibo Gao

**Comment by ZhiboGao** [comment]
Sorry, for the first question, the variable information is rain_forecast ([lat | 87] x [lon | 123]) and rain_obs ([lat | 87] x [lon | 123]). 
Thanks again for your help!

**Comment by JohnHalleyGotway** [comment, marked as answer]
Hello @ZhiboGao, I see you have question about how to format gridded NetCDF data to be read into the MET tools. MET does read a few different flavors of NetCDF files, but I always recommend that users format their gridded data following the NetCDF climate-forecast (CF) convention: https://cfconventions.org/

Listed on that page are some websites that check how closely your data file follows the CF-convention. And it's our goal that MET be able to read any gridded dataset following that convention.

I would say that users typically do NOT store both their forecast and observation data in the same file. I suppose you could, but I think you'll find it easier to have them in separate files. By way of example, I've listed below a sample NetCDF header for a file on a lat/lon grid that follows the CF-convention.

Some details to note are:
- lat, lon, and z are all coordinate variables (i.e. there's both dimensions and variables by that name which define their values).
- the time coordinate variable define the valid time for the data
- the forecast_reference_time specifies the model initialization time
- edr(time, z, lat, lon) defines the data variable... and the "grid_mapping" attribute defines the name of the variable that defines the grid information (in this case, that variable is also named grid_mapping)
- the global "Conventions = "CF-1.4" attribute identifies this file as being CF-compliant.
```
netcdf gtg_obs_forecast.20130827.i12.f06 {
dimensions:
	lat = 361 ;
	lon = 720 ;
	z = 26 ;
	name_len = 32 ;
	time = 1 ;
variables:
	int grid_mapping ;
		grid_mapping:grid_mapping_name = "latitude_longitude" ;
		grid_mapping:semi_major_axis = 6371000. ;
		grid_mapping:inverse_flattening = 0 ;
	double time(time) ;
		time:long_name = "forecast time" ;
		time:units = "seconds since 1970-1-1" ;
		time:axis = "T" ;
	double forecast_reference_time ;
		forecast_reference_time:long_name = "reference time" ;
		forecast_reference_time:units = "seconds since 1970-1-1" ;
	int forecast_period ;
		forecast_period:units = "seconds" ;
	double lat(lat) ;
		lat:units = "degrees_north" ;
		lat:long_name = "latitude coordinate" ;
		lat:standard_name = "latitude" ;
		lat:_CoordinateAxisType = "Lat" ;
	double lon(lon) ;
		lon:units = "degrees_east" ;
		lon:long_name = "longitude coordinate" ;
		lon:standard_name = "longitude" ;
		lon:_CoordinateAxisType = "Lon" ;
	double z(z) ;
		z:standard_name = "altitude" ;
		z:long_name = "flight_level" ;
		z:units = "feet" ;
		z:positive = "up" ;
		z:axis = "Z" ;
	int field_id ;
		field_id:long_name = "field indentification number" ;
	char field_name(name_len) ;
		field_name:long_name = "field name" ;
	float edr(time, z, lat, lon) ;
		edr:units = "m^(2/3) s^-1" ;
		edr:long_name = "Median eddy dissipation rate" ;
		edr:coordinates = "lat lon" ;
		edr:_FillValue = -9999.f ;
		edr:grid_mapping = "grid_mapping" ;

// global attributes:
		:Conventions = "CF-1.4" ;
		:Originating_center = "NCAR" ;
		:Product_Status = "Experimental Products" ;
		:Product_Type = "Forecast/Uninitialized Analysis" ;
		:DataType = "GRID" ;
		:DatasetLocation = "" ;
}
```
Whenever getting going with new datasets in MET, I always recommend starting by running them through plot_data_plane. For this example data file, you'd plot it by running:
```
plot_data_plane gtg_obs_forecast.20130827.i12.f06.nc plot.ps 'name="edr"; level="(0,0,*,*)";'
```
Then look at the resulting plot to make sure the data is oriented correctly and at the right spot on the earth.

Note that you could choose to write up a python script and pass the data to MET using python embedding. But that may (or may not) be more difficult to get up and running.

Hope that helps get you going.

**Reply by ZhiboGao** [reply]
Dear @JohnHalleyGotway,

Thank you so much for your prompt reply, it is of great help to my experiment. Wish you all the best !!

Kind regards
Zhibo Gao
